### DISCUSSION TITLE
Should I be seeing contact names in my HTML exports?

### DISCUSSION THREAD

**Discussion by DaveLinger** [ReagentX/imessage-exporter#64] [Q&A] [answered]
Seems like all of my HTML exports just have my contacts' phone number or email address listed, and not the contact's name. Is this the expected behavior?

**Comment by ReagentX** [comment]
See #62

**Comment by DaveLinger** [comment]
Got it. IMO, this would be more clear if the examples provided on the main page showed +15559091212 and johndoe@gmail.com or something instead of "Contact 1" and "Contact 2".

**Comment by ReagentX** [comment]
Yeah I had this thought too, it does feel a little misleading now. I will make that change!

**Comment by ReagentX** [comment, marked as answer]
Native support for contact detail resolution was merged via #631 and was released in [Desert Globemallow](https://github.com/ReagentX/imessage-exporter/releases/tag/3.2.0).

**Reply by dingdongwilly** [reply]
hi good stuff here with this tool....I have windows prebuilt installed version 3.2.1 and my output shows phone numbers instead of contacts..  see below.  99% of the messages backed up just show the phone number ie +1555-111-3333 format.

`C:\Users\Joe\Downloads>imessage-exporter-x86_64-pc-windows-gnu.exe -f html -c clone -p "C:\Users\Joe\AppData\Roaming\Apple Computer\MobileSync\Backup\00008130-001849300AF3803A" -x abc123
No HEIC converter found, image attachments will not be converted!
No CAF converter found, audio attachments will not be converted!
No MOV converter found, video attachments will not be converted!
Decrypting iOS backup...
  [1/5] Deriving backup keys...
  [2/5] Resolving messages database...
  [3/5] Decrypting messages database...
  [4/5] Resolving contacts database...
  [5/5] Decrypting contacts database...
Decrypted iOS backup: Nero2 (version 18.7.2)

Building cache...
  [1/5] Caching chats...
  [2/5] Caching chatrooms...
  [3/5] Caching participants...
  [4/5] Caching tapbacks...
  [5/5] Caching translations...
Cache built!
Estimated export size: 37.84 GB`

**Reply by ReagentX** [reply]
You can run with the diagnostic flag enabled to see how many contacts were matched with message handles. It is worth noting that file names use the original handle phone number/email to disambiguate output.

**Reply by dingdongwilly** [reply]
thanks...here is the output

`Cache built!

iMessage Database Diagnostics

Handle diagnostic data:
    Total handles: 1481
    Handles with more than one ID: 1
    Total duplicated handles: 325
Message diagnostic data:
    Total messages: 114448
    Messages not associated with a chat: 30
    Messages belonging to more than one chat: 13
Attachment diagnostic data:
    Total attachments: 17373
        Data referenced in table: 37.83 GB
        Data present on disk: 25.95 GB
    Missing files: 1146 (7%)
        No path provided: 1
        No file located: 1145
Thread diagnostic data:
    Total chats: 1427
    Total duplicated chats: 9
Global diagnostic data:
    Total database size: 166.16 MB
    Handles with resolved names: 407/1156 (35%)`

**Reply by dingdongwilly** [reply]
these are all text/ iMessage conversations and it's pretty rare I text people without a contact object in my iphone

**Reply by ReagentX** [reply]
Looks like about 1/3 of the handles matched. Again, as noted in the release notes, to uniquely identify each file, file names use the original handle details.

**Reply by dingdongwilly** [reply]
found my issue, majority of my iphone contacts were associated with gmail account, only a small amount were on iphone.  I moved all of them to icloud and synced with iphone and just took another itunes backup, running the imessage export again to see if I get better hit ratio for contacts in the html output

**Reply by dingdongwilly** [reply]
update - I removed my google / gmail contacts on iphone, merged them all to iCloud and confirmed they synced with iPhone.   Ran a new backup in iTunes, ran diag mode for iexplorere and its still only finding a small %, see below....

Handles with resolved names: 401/1156 (35%)

so the output in html looks as follows, which I am wondering is this supposed to resolve these numbers to my contacts name?
<img width="568" height="500" alt="Screenshot 2025-11-28 145306" src="https://github.com/user-attachments/assets/423ef209-e838-4853-87e8-4e941c16e123" />

**Reply by ReagentX** [reply]
To uniquely identify each file, file names use the original handle details. This prevents duplicate filenames when multiple contacts share the same name. Resolved handle details are used in the exports.

**Reply by dingdongwilly** [reply]
ok thanks, just so I am clear then, there is no way to get these numbers to resolve with your (great) tool?

**Reply by ReagentX** [reply]
The resolved contact details are present in the exported files. Filenames use the original handle details.

**Reply by dingdongwilly** [reply]
ah ok, thank you
